sdk/search/search-documents/test/utils/setup.ts

```diff
@@ -421,8 +421,25 @@ export async function populateIndex(client: SearchClient<Hotel>): Promise<void>
 
 // eslint-disable-next-line @azure/azure-sdk/ts-use-interface-parameters
 export async function createDataSourceConnections(client: SearchIndexerClient): Promise<void> {
+  const testCaseNames: string[] = ["my-data-source-1", "my-data-source-2"];
+  const dataSourceConnectionNames: string[] = await client.listDataSourceConnectionsNames();
+  const unCommonElements: string[] = dataSourceConnectionNames.filter(
+    (element) => !testCaseNames.includes(element)
+  );
+  if (unCommonElements.length > 0) {
+    // There are datasource connections which are already existing in this subscription.
+    // We do not want to delete them by accident. So, we are returning without further
+    // action. The test cases will fail. Please do not use a subscription which already
+    // has datasource connections for testing.
+    return;
```

Comment:
Is there some way to indicate to the person running the test why this failed so they know what they have to do to run it successfully? I feel like this could be hard to debug. 

Ideally the test would fail with a message indicating what the problem is instead of the resource silently not being created correctly.  I certainly wouldn't want to try to debug CI logs of this happening.

Reply:
Added assert fail with detailed message in each case.